### PR TITLE
Move multi-rename from sandbox/ into scripts; add line length limits. #450

### PR DESCRIPTION
**Pull request by SherineAwad**
add line length limits and move multi-rename from sandbox/ into scripts;  #450

This is needed in khmer-protocols to remove the fold step.

### CODE REVIEW

**Comment by SherineAwad** [comment]
Ready for review !

 Is it mergable?
 Did it pass the tests?
 If it introduces new functionality in scripts/ is it tested? Check for code coverage. Is it well formatted? Look at make pep8, make diff_pylint_report, make cppcheck, and make doc output. Use make format and manual fixing as needed.
 Did it change the command-line interface? Only additions are allowed without a major version increment. Changing file formats also requires a major version number increment.
 Is it documented in the ChangeLog?
 Was a spellchecker run on the source code and documentation after changes were made?

**Comment by SherineAwad** [comment]
Ready for review
- [x] Is it mergable?
- [x] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage.
- [x] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Is it documented in the ChangeLog?
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

File: scripts/multi-rename.py [hunk 1]
```diff
@@ -19,8 +20,9 @@ def main():
         for record in screed.open(filename):
             if len(record.sequence) >= CUTOFF:
                 n += 1
-                print('>%s.%s %s\n%s' % (prefix, n, record.name, record.sequence))
-
+                print('>%s.%s %s' % (prefix, n, record.name))
+                x = "\n".join(textwrap.wrap(record.sequence, 80))
+                print ('%s' % (x))
```
Comment (ctb) [inline scripts/multi-rename.py]: you can just do print(x) here.

File: tests/test_scripts.py [hunk 2]
```diff
@@ -60,7 +60,7 @@ def test_load_into_counting_max_memory_usage_parameter():
 
     outfile = utils.get_temp_filename('out.ct')
     infile = utils.get_test_data('test-abund-read-2.fa')
-
```
Comment (ctb) [inline tests/test_scripts.py]: this change is unnecessary and introduces extraneous spacing, pls remove

File: tests/test_scripts.py [hunk 3]
```diff
@@ -3043,7 +3043,7 @@ def test_trim_low_abund_highfpr():
 def test_trim_low_abund_trimtest():
     infile = utils.get_temp_filename('test.fa')
     in_dir = os.path.dirname(infile)
-
```
Comment (ctb) [inline tests/test_scripts.py]: same comment - pls remove

File: tests/test_scripts.py [hunk 4]
```diff
@@ -3218,3 +3218,17 @@ def test_roundtrip_commented_format():
...
+    infile1 = utils.get_temp_filename('test-multi.fa')
+    multioutfile = utils.get_temp_filename('out.fa')
+    infile2 = utils.get_temp_filename('out.fa')
+    shutil.copyfile(utils.get_test_data('test-multi.fa'), infile1)
+    shutil.copyfile(utils.get_test_data('multi-output.fa'), infile2)
+    args = ['assembly', infile1]
+    _, out, err = utils.runscript('multi-rename.py', args)
+    r = open(infile2).read()
+    for k in r:
+        assert k in out, (k, out)
```
Comment (ctb) [inline tests/test_scripts.py]: this assertion is checking that every character in 'r' in present somewhere in 'out', which is probably not what you want?

if you are trying to compare multi-output.fa to the output of multi-rename, you probably want to do:

`assert r == out`

or some such.

File: ChangeLog [hunk 5]
```diff
@@ -1,3 +1,9 @@
+2015-07-04  Sherine Awad  <drmahmoud@ucdavis.edu>
```
Comment (ctb) [inline ChangeLog]: the format of your ChangeLog entry doesn't match the other entries; please fix.

File: tests/test_scripts.py (hunk 2 above)
Comment (ctb) [inline tests/test_scripts.py]: ping @SherineAwad

File: tests/test_scripts.py (hunk 3 above)
Comment (ctb) [inline tests/test_scripts.py]: ping @SherineAwad

File: ChangeLog [hunk 6]
```diff
@@ -1,3 +1,9 @@
+2015-07-27  Sherine Awad  <drmahmoud@ucdavis.edu>
+
+   * Add a line break in multi-rename.py
```
Comment (ctb) [inline ChangeLog]: Please fix this format to look like the other ChangeLog entries: for each line, '*', then filename, followed by colon, followed by description.

**Comment by ctb** [comment]
When you're ready for another review, please ping me with '@ctb'

**Comment by SherineAwad** [comment]
@CTB Ready For Review

**Comment by ctb** [comment]
Notes about your ChangeLog diff:

* one entry per file changed, generally;
* use of 'multirename.py' (should be 'multi-rename.py')

I'll fix.

**Comment by ctb** [comment]
oh, and, you don't need to repeat the name of the script in the details!

**Comment by ctb** [comment]
Whoops, I failed to understand that this was moving multi-rename.py into scripts/.  We have a set of requirements for things in scripts/; I've pasted in the info here:

https://github.com/dib-lab/khmer/issues/450#issuecomment-126915712

For now, I'll merge this with the script in sandbox/. If you want to pursue moving it into scripts/, please do go ahead! And apologies for missing this in all the previous rounds of review.

**Pull request merged**